[neeo] Can Not Add Brain - OH3

I'm running a fresh install of OH3 M2 and trying to migrate from OH2.  I have no issues on OH2 with the neeo binding.  When I try to add a brain on OH 3 I get the following error when trying to add a brain.

```
2020-11-08 18:50:13.177 [WARN ] [org.eclipse.jetty.server.HttpChannel] - /neeo/neeostatus/addbrain
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.io.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.io.neeo.internal.NeeoApi.getSystemInfo(NeeoApi.java:204) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.addDiscovered(MdnsBrainDiscovery.java:302) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.addDiscovered(MdnsBrainDiscovery.java:287) ~[?:?]
        at org.openhab.io.neeo.NeeoService.addBrain(NeeoService.java:526) ~[?:?]
        at org.openhab.io.neeo.internal.servletservices.BrainDashboardService.handlePost(BrainDashboardService.java:166) ~[?:?]
        at org.openhab.io.neeo.internal.AbstractServlet.doPost(AbstractServlet.java:151) ~[?:?]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) ~[bundleFile:3.1.0]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[bundleFile:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:852) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:544) ~[bundleFile:9.4.20.v20190813]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:536) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1581) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1307) ~[bundleFile:9.4.20.v20190813]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:293) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1549) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1204) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[bundleFile:9.4.20.v20190813]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.Server.handle(Server.java:494) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [bundleFile:9.4.20.v20190813]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 43 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 43 more

```

Also on start I get:

```
2020-11-08 19:09:47.874 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:47) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:167) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:146) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more

```

Ok lots of info, let me try to break this out.

Installed the snapshot binding from J-N-K:
openhab> bundle:list -s | grep -i neeo
208 │ Active │  80 │ 3.0.0.202011111831      │ org.openhab.binding.neeo
261 │ Active │  80 │ 3.0.0.M2                │ org.openhab.io.neeo

Info as requested:
openhab> find-class javax.ws.rs.client.ClientBuilder

Apache Aries JAX-RS Specification API (45)
javax/ws/rs/client/ClientBuilder.class

Apache Aries JAX-RS Whiteboard (46)
javax/ws/rs/client/ClientBuilder.class

org.osgi:org.osgi.service.jaxrs (200)
javax/ws/rs/client/ClientBuilder.class

openHAB Add-ons :: Bundles :: Neeo Binding (208)
javax/ws/rs/client/ClientBuilder.class

openHAB Add-ons :: Bundles :: Remote openHAB Binding (237)
javax/ws/rs/client/ClientBuilder.class

openHAB Add-ons :: Bundles :: IO :: NEEO Integration (261)
javax/ws/rs/client/ClientBuilder.class
openhab> list | grep JAX-RS
 45 │ Active │  80 │ 1.0.1                   │ Apache Aries JAX-RS Specification API
 46 │ Active │  80 │ 1.0.9                   │ Apache Aries JAX-RS Whiteboard


Still no fix on the issue.  Restarted a few times.  Cleared the cache and restarted a few times.

On start:
2020-11-11 19:48:07.690 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:69) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:172) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:151) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
2020-11-11 19:48:07.697 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:69) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:172) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:151) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
2020-11-11 19:48:16.625 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.io.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.io.neeo.internal.NeeoApi.getSystemInfo(NeeoApi.java:204) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.considerService(MdnsBrainDiscovery.java:253) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.considerService(MdnsBrainDiscovery.java:219) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.lambda$0(MdnsBrainDiscovery.java:156) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 12 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 12 more

When I try to add a brain
2020-11-11 19:49:07.357 [WARN ] [org.eclipse.jetty.server.HttpChannel] - /neeo/neeostatus/addbrain
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.io.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.io.neeo.internal.NeeoApi.getSystemInfo(NeeoApi.java:204) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.addDiscovered(MdnsBrainDiscovery.java:302) ~[?:?]
        at org.openhab.io.neeo.internal.discovery.MdnsBrainDiscovery.addDiscovered(MdnsBrainDiscovery.java:287) ~[?:?]
        at org.openhab.io.neeo.NeeoService.addBrain(NeeoService.java:526) ~[?:?]
        at org.openhab.io.neeo.internal.servletservices.BrainDashboardService.handlePost(BrainDashboardService.java:166) ~[?:?]
        at org.openhab.io.neeo.internal.AbstractServlet.doPost(AbstractServlet.java:151) ~[?:?]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:707) ~[bundleFile:3.1.0]
        at javax.servlet.http.HttpServlet.service(HttpServlet.java:790) ~[bundleFile:3.1.0]
        at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:852) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:544) ~[bundleFile:9.4.20.v20190813]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceServletHandler.doHandle(HttpServiceServletHandler.java:71) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:536) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1581) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1307) ~[bundleFile:9.4.20.v20190813]
        at org.ops4j.pax.web.service.jetty.internal.HttpServiceContext.doHandle(HttpServiceContext.java:293) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:482) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1549) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1204) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141) ~[bundleFile:9.4.20.v20190813]
        at org.ops4j.pax.web.service.jetty.internal.JettyServerHandlerCollection.handle(JettyServerHandlerCollection.java:80) ~[bundleFile:?]
        at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.Server.handle(Server.java:494) ~[bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:374) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:268) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:367) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:782) [bundleFile:9.4.20.v20190813]
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:918) [bundleFile:9.4.20.v20190813]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 43 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 43 more

I ran it again just to be sure...

openhab> list org.openhab.binding.neeo
START LEVEL 100 , List Threshold: 50
 ID │ State  │ Lvl │ Version            │ Name
────┼────────┼─────┼────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
208 │ Active │  80 │ 3.0.0.202011111831 │ openHAB Add-ons :: Bundles :: Neeo Binding
openhab> update org.openhab.binding.neeo https://janessa.me/esh/org.openhab.binding.neeo-3.0.0-SNAPSHOT.jar
openhab> list org.openhab.binding.neeo
START LEVEL 100 , List Threshold: 50
 ID │ State  │ Lvl │ Version            │ Name
────┼────────┼─────┼────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
208 │ Active │  80 │ 3.0.0.202011111831 │ openHAB Add-ons :: Bundles :: Neeo Binding

And on restart...

2020-11-11 20:17:52.136 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:69) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:172) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:151) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
2020-11-11 20:17:52.139 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:69) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:172) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:151) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more

Just as a test, I removed the IO completely and I still get the first two errors on startup.  I do not get the third error anymore (which makes sense because the first two showed on binding.neeo and the third was on io.neeo).

openhab> list org.openhab.io.neeo
START LEVEL 100 , List Threshold: 50
ID │ State │ Lvl │ Version │ Name
───┼───────┼─────┼─────────┼─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
openhab> list org.openhab.binding.neeo
START LEVEL 100 , List Threshold: 50
 ID │ State  │ Lvl │ Version            │ Name
────┼────────┼─────┼────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
208 │ Active │  80 │ 3.0.0.202011111831 │ openHAB Add-ons :: Bundles :: Neeo Binding
openhab>

2020-11-11 20:37:04.872 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:69) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:172) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:151) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
2020-11-11 20:37:04.879 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception:
java.lang.RuntimeException: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:50) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:55) ~[?:?]
        at org.openhab.binding.neeo.internal.net.HttpRequest.<init>(HttpRequest.java:51) ~[?:?]
        at org.openhab.binding.neeo.internal.NeeoBrainApi.<init>(NeeoBrainApi.java:69) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.initializeTask(NeeoBrainHandler.java:172) ~[?:?]
        at org.openhab.binding.neeo.internal.handler.NeeoBrainHandler.lambda$0(NeeoBrainHandler.java:151) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:120) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more
Caused by: java.lang.InstantiationException
        at jdk.internal.reflect.InstantiationExceptionConstructorAccessorImpl.newInstance(InstantiationExceptionConstructorAccessorImpl.java:48) ~[?:?]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[?:?]
        at java.lang.Class.newInstance(Class.java:584) ~[?:?]
        at javax.ws.rs.client.ClientFinder.newInstance(ClientFinder.java:116) ~[?:?]
        at javax.ws.rs.client.ClientFinder.find(ClientFinder.java:96) ~[?:?]
        at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:44) ~[?:?]
        ... 11 more

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/NeeoBrainApi.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/NeeoBrainApi.java
@@ -17,6 +17,8 @@ import java.net.URL;
 import java.util.Objects;
 import java.util.concurrent.atomic.AtomicReference;
 
+import javax.ws.rs.client.ClientBuilder;
+
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jetty.http.HttpStatus;
 import org.openhab.binding.neeo.internal.models.ExecuteResult;
@@ -44,7 +46,10 @@ public class NeeoBrainApi implements AutoCloseable {
     private final Gson gson = NeeoUtil.getGson();
 
     /** The {@link HttpRequest} used for making requests */
-    private final AtomicReference<HttpRequest> request = new AtomicReference<>(new HttpRequest());
+    private final AtomicReference<HttpRequest> request;
+
+    /** The {@link ClientBuilder} to use */
+    private final ClientBuilder clientBuilder;
 
     /** The IP address of the neeo brain */
     private final NeeoUrlBuilder urlBuilder;
@@ -54,11 +59,14 @@ public class NeeoBrainApi implements AutoCloseable {
      *
      * @param ipAddress the non-empty ip address
      */
-    public NeeoBrainApi(String ipAddress) {
+    public NeeoBrainApi(String ipAddress, ClientBuilder clientBuilder) {
         NeeoUtil.requireNotEmpty(ipAddress, "ipAddress cannot be empty");
 
         this.urlBuilder = new NeeoUrlBuilder(
                 NeeoConstants.PROTOCOL + ipAddress + ":" + NeeoConstants.DEFAULT_BRAIN_PORT);
+        this.clientBuilder = clientBuilder;
+
+        request = new AtomicReference<>(new HttpRequest(clientBuilder));
     }
 
     /**
@@ -232,7 +240,7 @@ public class NeeoBrainApi implements AutoCloseable {
 
     @Override
     public void close() throws Exception {
-        NeeoUtil.close(request.getAndSet(new HttpRequest()));
+        NeeoUtil.close(request.getAndSet(new HttpRequest(clientBuilder)));
     }
 
     /**

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoBrainHandler.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoBrainHandler.java
@@ -189,19 +189,18 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
             if (config.isEnableForwardActions()) {
                 NeeoUtil.checkInterrupt();
 
-                forwardActionServlet = new NeeoForwardActionsServlet(scheduler,
-                        json -> {
-                            triggerChannel(NeeoConstants.CHANNEL_BRAIN_FOWARDACTIONS, json);
+                forwardActionServlet = new NeeoForwardActionsServlet(scheduler, json -> {
+                    triggerChannel(NeeoConstants.CHANNEL_BRAIN_FOWARDACTIONS, json);
 
-                            final NeeoAction action = Objects.requireNonNull(gson.fromJson(json, NeeoAction.class));
+                    final NeeoAction action = Objects.requireNonNull(gson.fromJson(json, NeeoAction.class));
 
-                            for (final Thing child : getThing().getThings()) {
-                                final ThingHandler th = child.getHandler();
-                                if (th instanceof NeeoRoomHandler) {
-                                    ((NeeoRoomHandler) th).processAction(action);
-                                }
-                            }
-                        }, config.getForwardChain(), clientBuilder);
+                    for (final Thing child : getThing().getThings()) {
+                        final ThingHandler th = child.getHandler();
+                        if (th instanceof NeeoRoomHandler) {
+                            ((NeeoRoomHandler) th).processAction(action);
+                        }
+                    }
+                }, config.getForwardChain(), clientBuilder);
 
                 NeeoUtil.checkInterrupt();
                 try {

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoBrainHandler.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoBrainHandler.java
@@ -29,6 +29,7 @@ import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
 import javax.servlet.ServletException;
+import javax.ws.rs.client.ClientBuilder;
 
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
@@ -73,6 +74,9 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
     /** The {@link NetworkAddressService} to use */
     private final NetworkAddressService networkAddressService;
 
+    /** The {@link ClientBuilder} to use */
+    private final ClientBuilder clientBuilder;
+
     /** GSON implementation - only used to deserialize {@link NeeoAction} */
     private final Gson gson = new Gson();
 
@@ -112,7 +116,7 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
      * @param networkAddressService the non-null {@link NetworkAddressService}
      */
     NeeoBrainHandler(Bridge bridge, int servicePort, HttpService httpService,
-            NetworkAddressService networkAddressService) {
+            NetworkAddressService networkAddressService, ClientBuilder clientBuilder) {
         super(bridge);
 
         Objects.requireNonNull(bridge, "bridge cannot be null");
@@ -122,6 +126,7 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
         this.servicePort = servicePort;
         this.httpService = httpService;
         this.networkAddressService = networkAddressService;
+        this.clientBuilder = clientBuilder;
     }
 
     /**
@@ -164,7 +169,7 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
                         "Brain IP Address must be specified");
                 return;
             }
-            final NeeoBrainApi api = new NeeoBrainApi(ipAddress);
+            final NeeoBrainApi api = new NeeoBrainApi(ipAddress, clientBuilder);
             final NeeoBrain brain = api.getBrain();
             final String brainId = getNeeoBrainId();
 
@@ -199,7 +204,7 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
                                     }
                                 }
                             }
-                        }, config.getForwardChain());
+                        }, config.getForwardChain(), clientBuilder);
 
                 NeeoUtil.checkInterrupt();
                 try {

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoBrainHandler.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoBrainHandler.java
@@ -190,18 +190,15 @@ public class NeeoBrainHandler extends BaseBridgeHandler {
                 NeeoUtil.checkInterrupt();
 
                 forwardActionServlet = new NeeoForwardActionsServlet(scheduler,
-                        new NeeoForwardActionsServlet.Callback() {
-                            @Override
-                            public void post(String json) {
-                                triggerChannel(NeeoConstants.CHANNEL_BRAIN_FOWARDACTIONS, json);
+                        json -> {
+                            triggerChannel(NeeoConstants.CHANNEL_BRAIN_FOWARDACTIONS, json);
 
-                                final NeeoAction action = Objects.requireNonNull(gson.fromJson(json, NeeoAction.class));
+                            final NeeoAction action = Objects.requireNonNull(gson.fromJson(json, NeeoAction.class));
 
-                                for (final Thing child : getThing().getThings()) {
-                                    final ThingHandler th = child.getHandler();
-                                    if (th instanceof NeeoRoomHandler) {
-                                        ((NeeoRoomHandler) th).processAction(action);
-                                    }
+                            for (final Thing child : getThing().getThings()) {
+                                final ThingHandler th = child.getHandler();
+                                if (th instanceof NeeoRoomHandler) {
+                                    ((NeeoRoomHandler) th).processAction(action);
                                 }
                             }
                         }, config.getForwardChain(), clientBuilder);

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoForwardActionsServlet.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoForwardActionsServlet.java
@@ -20,6 +20,7 @@ import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.client.ClientBuilder;
 
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang.StringUtils;
@@ -52,6 +53,9 @@ public class NeeoForwardActionsServlet extends HttpServlet {
     @Nullable
     private final String forwardChain;
 
+    /** The {@link ClientBuilder} to use */
+    private final ClientBuilder clientBuilder;
+
     /** The scheduler to use to schedule recipe execution */
     private final ScheduledExecutorService scheduler;
 
@@ -62,7 +66,8 @@ public class NeeoForwardActionsServlet extends HttpServlet {
      * @param callback a non-null {@link Callback}
      * @param forwardChain a possibly null, possibly empty forwarding chain
      */
-    NeeoForwardActionsServlet(ScheduledExecutorService scheduler, Callback callback, @Nullable String forwardChain) {
+    NeeoForwardActionsServlet(ScheduledExecutorService scheduler, Callback callback, @Nullable String forwardChain,
+            ClientBuilder clientBuilder) {
         super();
 
         Objects.requireNonNull(scheduler, "scheduler cannot be null");
@@ -71,6 +76,7 @@ public class NeeoForwardActionsServlet extends HttpServlet {
         this.scheduler = scheduler;
         this.callback = callback;
         this.forwardChain = forwardChain;
+        this.clientBuilder = clientBuilder;
     }
 
     /**
@@ -94,7 +100,7 @@ public class NeeoForwardActionsServlet extends HttpServlet {
         final String fc = forwardChain;
         if (fc != null && StringUtils.isNotEmpty(fc)) {
             scheduler.execute(() -> {
-                try (final HttpRequest request = new HttpRequest()) {
+                try (final HttpRequest request = new HttpRequest(clientBuilder)) {
                     for (final String forwardUrl : fc.split(",")) {
                         if (StringUtils.isNotEmpty(forwardUrl)) {
                             final HttpResponse httpResponse = request.sendPostJsonCommand(forwardUrl, json);

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoForwardActionsServlet.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoForwardActionsServlet.java
@@ -16,14 +16,12 @@ import java.io.IOException;
 import java.util.Objects;
 import java.util.concurrent.ScheduledExecutorService;
 
-import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.ws.rs.client.ClientBuilder;
 
 import org.apache.commons.io.IOUtils;
-import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.jetty.http.HttpStatus;
@@ -87,10 +85,11 @@ public class NeeoForwardActionsServlet extends HttpServlet {
      * @param resp the non-null response
      */
     @Override
-    protected void doPost(@Nullable HttpServletRequest req, @Nullable HttpServletResponse resp)
-            throws ServletException, IOException {
-        Objects.requireNonNull(req, "req cannot be null");
-        Objects.requireNonNull(resp, "resp cannot be null");
+    protected void doPost(@Nullable HttpServletRequest req, @Nullable HttpServletResponse resp) throws IOException {
+        if (req == null || resp == null) {
+            logger.warn("doPost called with req={}, resp={}, non-null required.", req, resp);
+            return;
+        }
 
         final String json = IOUtils.toString(req.getReader());
         logger.debug("handleForwardActions {}", json);
@@ -98,11 +97,11 @@ public class NeeoForwardActionsServlet extends HttpServlet {
         callback.post(json);
 
         final String fc = forwardChain;
-        if (fc != null && StringUtils.isNotEmpty(fc)) {
+        if (fc != null && !fc.isEmpty()) {
             scheduler.execute(() -> {
                 try (final HttpRequest request = new HttpRequest(clientBuilder)) {
                     for (final String forwardUrl : fc.split(",")) {
-                        if (StringUtils.isNotEmpty(forwardUrl)) {
+                        if (forwardUrl != null && !forwardUrl.isEmpty()) {
                             final HttpResponse httpResponse = request.sendPostJsonCommand(forwardUrl, json);
                             if (httpResponse.getHttpCode() != HttpStatus.OK_200) {
                                 logger.debug("Cannot forward event {} to {}: {}", json, forwardUrl,

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoHandlerFactory.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/handler/NeeoHandlerFactory.java
@@ -17,6 +17,8 @@ import java.util.Objects;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
+import javax.ws.rs.client.ClientBuilder;
+
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
 import org.openhab.binding.neeo.internal.NeeoConstants;
@@ -33,6 +35,7 @@ import org.openhab.core.thing.binding.BaseThingHandlerFactory;
 import org.openhab.core.thing.binding.ThingHandler;
 import org.openhab.core.thing.binding.ThingHandlerFactory;
 import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Activate;
 import org.osgi.service.component.annotations.Component;
 import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.http.HttpService;
@@ -48,54 +51,23 @@ import org.osgi.service.http.HttpService;
 public class NeeoHandlerFactory extends BaseThingHandlerFactory {
 
     /** The {@link HttpService} used to register callbacks */
-    @NonNullByDefault({})
-    private HttpService httpService;
+    private final HttpService httpService;
 
     /** The {@link NetworkAddressService} used for ip lookup */
-    @NonNullByDefault({})
-    private NetworkAddressService networkAddressService;
+    private final NetworkAddressService networkAddressService;
+
+    /** The {@link ClientBuilder} used in HttpRequest */
+    private final ClientBuilder clientBuilder;
 
     /** The discovery services created by this class (one per room and one for each device) */
     private final ConcurrentMap<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new ConcurrentHashMap<>();
 
-    /**
-     * Sets the {@link HttpService}.
-     *
-     * @param httpService the non-null {@link HttpService} to use
-     */
-    @Reference
-    protected void setHttpService(HttpService httpService) {
-        Objects.requireNonNull(httpService, "httpService cannot be null");
+    @Activate
+    public NeeoHandlerFactory(@Reference HttpService httpService,
+            @Reference NetworkAddressService networkAddressService, @Reference ClientBuilder clientBuilder) {
         this.httpService = httpService;
-    }
-
-    /**
-     * Unsets the {@link HttpService}
-     *
-     * @param httpService the {@link HttpService} (not used in this implementation)
-     */
-    protected void unsetHttpService(HttpService httpService) {
-        this.httpService = null;
-    }
-
-    /**
-     * Sets the {@link NetworkAddressService}.
-     *
-     * @param networkAddressService the non-null {@link NetworkAddressService} to use
-     */
-    @Reference
-    protected void setNetworkAddressService(NetworkAddressService networkAddressService) {
-        Objects.requireNonNull(networkAddressService, "networkAddressService cannot be null");
         this.networkAddressService = networkAddressService;
-    }
-
-    /**
-     * Unsets the {@link NetworkAddressService}
-     *
-     * @param networkAddressService the {@link NetworkAddressService} (not used in this implementation)
-     */
-    protected void unsetNetworkAddressService(NetworkAddressService networkAddressService) {
-        this.networkAddressService = null;
+        this.clientBuilder = clientBuilder;
     }
 
     @Override
@@ -112,17 +84,10 @@ public class NeeoHandlerFactory extends BaseThingHandlerFactory {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
         if (thingTypeUID.equals(NeeoConstants.BRIDGE_TYPE_BRAIN)) {
-            final HttpService localHttpService = httpService;
-            final NetworkAddressService localNetworkAddressService = networkAddressService;
-
-            Objects.requireNonNull(localHttpService, "HttpService cannot be null");
-            Objects.requireNonNull(localNetworkAddressService, "networkAddressService cannot be null");
-
             final int port = HttpServiceUtil.getHttpServicePort(this.bundleContext);
-
             final NeeoBrainHandler handler = new NeeoBrainHandler((Bridge) thing,
-                    port < 0 ? NeeoConstants.DEFAULT_BRAIN_HTTP_PORT : port, localHttpService,
-                    localNetworkAddressService);
+                    port < 0 ? NeeoConstants.DEFAULT_BRAIN_HTTP_PORT : port, httpService, networkAddressService,
+                    clientBuilder);
             registerRoomDiscoveryService(handler);
             return handler;
         } else if (thingTypeUID.equals(NeeoConstants.BRIDGE_TYPE_ROOM)) {

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
@@ -13,6 +13,8 @@
 package org.openhab.binding.neeo.internal.net;
 
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.Objects;
 
 import javax.ws.rs.ProcessingException;
@@ -81,18 +83,23 @@ public class HttpRequest implements AutoCloseable {
     /**
      * Send post JSON command using the body
      *
-     * @param uri the non empty uri
+     * @param uriString the non empty uri
      * @param body the non-null, possibly empty body
      * @return the {@link HttpResponse}
      */
-    public HttpResponse sendPostJsonCommand(String uri, String body) {
-        NeeoUtil.requireNotEmpty(uri, "uri cannot be empty");
+    public HttpResponse sendPostJsonCommand(String uriString, String body) {
+        NeeoUtil.requireNotEmpty(uriString, "uri cannot be empty");
         Objects.requireNonNull(body, "body cannot be null");
 
-        logger.trace("sendPostJsonCommand: target={}, body={}", uri, body);
+        logger.trace("sendPostJsonCommand: target={}, body={}", uriString, body);
 
         try {
-            final Builder request = client.target(uri).request(MediaType.APPLICATION_JSON);
+            URI targetUri = new URI(uriString);
+            if (!targetUri.isAbsolute()) {
+                logger.warn("Absolute URI required but provided URI '{}' is non-absolute. ", uriString);
+                return new HttpResponse(HttpStatus.NOT_ACCEPTABLE_406, "Absolute URI required");
+            }
+            final Builder request = client.target(targetUri).request(MediaType.APPLICATION_JSON);
 
             final Response content = request.post(Entity.entity(body, MediaType.APPLICATION_JSON));
 
@@ -105,6 +112,8 @@ public class HttpRequest implements AutoCloseable {
             // as well
         } catch (IOException | IllegalStateException | IllegalArgumentException | ProcessingException e) {
             return new HttpResponse(HttpStatus.SERVICE_UNAVAILABLE_503, e.getMessage());
+        } catch (URISyntaxException e) {
+            return new HttpResponse(HttpStatus.NOT_ACCEPTABLE_406, e.getMessage());
         }
     }
 

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
@@ -47,8 +47,8 @@ public class HttpRequest implements AutoCloseable {
     /**
      * Instantiates a new request
      */
-    public HttpRequest() {
-        client = ClientBuilder.newClient();
+    public HttpRequest(ClientBuilder clientBuilder) {
+        client = clientBuilder.newClient();
 
         if (logger.isDebugEnabled()) {
             client.register(new LoggingFilter(new Slf4LoggingAdapter(logger), true));

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
@@ -89,6 +89,8 @@ public class HttpRequest implements AutoCloseable {
         NeeoUtil.requireNotEmpty(uri, "uri cannot be empty");
         Objects.requireNonNull(body, "body cannot be null");
 
+        logger.trace("sendPostJsonCommand: target={}, body={}", uri, body);
+
         try {
             final Builder request = client.target(uri).request(MediaType.APPLICATION_JSON);
 

--- a/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
+++ b/bundles/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/net/HttpRequest.java
@@ -48,7 +48,7 @@ public class HttpRequest implements AutoCloseable {
      * Instantiates a new request
      */
     public HttpRequest(ClientBuilder clientBuilder) {
-        client = clientBuilder.newClient();
+        client = clientBuilder.build();
 
         if (logger.isDebugEnabled()) {
             client.register(new LoggingFilter(new Slf4LoggingAdapter(logger), true));

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/NeeoService.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/NeeoService.java
@@ -23,6 +23,7 @@ import java.util.Set;
 import java.util.concurrent.CopyOnWriteArrayList;
 
 import javax.servlet.ServletException;
+import javax.ws.rs.client.ClientBuilder;
 
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
@@ -82,19 +83,20 @@ public class NeeoService implements EventSubscriber, NetworkAddressChangeListene
      * This is the context created in the activate method (and nulled in the deactivate method) that will provide the
      * context to all services for all servlets
      */
-    private @Nullable ServiceContext context;
+    private final ServiceContext context;
 
     // The following services are set by openHAB via the getter/setters
-    private @Nullable HttpService httpService;
-    private @Nullable ItemRegistry itemRegistry;
-    private @Nullable BindingInfoRegistry bindingInfoRegistry;
-    private @Nullable ThingRegistry thingRegistry;
-    private @Nullable ThingTypeRegistry thingTypeRegistry;
-    private @Nullable ItemChannelLinkRegistry itemChannelLinkRegistry;
-    private @Nullable ChannelTypeRegistry channelTypeRegistry;
-    private @Nullable MDNSClient mdnsClient;
-    private @Nullable EventPublisher eventPublisher;
-    private @Nullable NetworkAddressService networkAddressService;
+    private final HttpService httpService;
+    private final ItemRegistry itemRegistry;
+    private final BindingInfoRegistry bindingInfoRegistry;
+    private final ThingRegistry thingRegistry;
+    private final ThingTypeRegistry thingTypeRegistry;
+    private final ItemChannelLinkRegistry itemChannelLinkRegistry;
+    private final ChannelTypeRegistry channelTypeRegistry;
+    private final MDNSClient mdnsClient;
+    private final EventPublisher eventPublisher;
+    private final NetworkAddressService networkAddressService;
+    private final ClientBuilder clientBuilder;
 
     /** The main dashboard servlet. Only created in the activate method (and disposed of in the deactivate method) */
     private @Nullable NeeoDashboardServlet dashboardServlet;
@@ -136,238 +138,24 @@ public class NeeoService implements EventSubscriber, NetworkAddressChangeListene
         }
     };
 
-    /**
-     * The event filter to apply to this service
-     */
-    private final EventFilter eventFilter = event -> {
-        logger.trace("apply: {}", event);
-
-        for (NeeoBrainServlet ns : servlets) {
-            final List<EventFilter> efs = ns.getEventFilters();
-            if (efs != null) {
-                for (EventFilter ef : efs) {
-                    if (ef.apply(event)) {
-                        logger.trace("apply (true): {}", event);
-                        return true;
-                    }
-                }
-            }
-        }
-
-        logger.trace("apply (false): {}", event);
-        return false;
-    };
-
-    /**
-     * Sets the http service.
-     *
-     * @param httpService the non-null http service
-     */
-    @Reference
-    public void setHttpService(HttpService httpService) {
-        Objects.requireNonNull(httpService, "httpService cannot be null");
-        this.httpService = httpService;
-    }
-
-    /**
-     * Unset http service.
-     *
-     * @param httpService the http service (ignored)
-     */
-    public void unsetHttpService(HttpService httpService) {
-        this.httpService = null;
-    }
-
-    /**
-     * Sets the item registry.
-     *
-     * @param itemRegistry the non-null item registry
-     */
-    @Reference
-    public void setItemRegistry(ItemRegistry itemRegistry) {
-        Objects.requireNonNull(itemRegistry, "itemRegistry cannot be null");
-        this.itemRegistry = itemRegistry;
-    }
-
-    /**
-     * Unset item registry.
-     *
-     * @param itemRegistry the item registry (ignored)
-     */
-    public void unsetItemRegistry(ItemRegistry itemRegistry) {
-        this.itemRegistry = null;
-    }
-
-    /**
-     * Sets the binding info registry.
-     *
-     * @param bindingInfoRegistry the non-null binding info registry
-     */
-    @Reference
-    public void setBindingInfoRegistry(BindingInfoRegistry bindingInfoRegistry) {
-        Objects.requireNonNull(bindingInfoRegistry, "bindingInfoRegistry cannot be null");
-        this.bindingInfoRegistry = bindingInfoRegistry;
-    }
-
-    /**
-     * Unset binding info registry.
-     *
-     * @param bindingInfoRegistry the binding info registry (ignored)
-     */
-    public void unsetBindingInfoRegistry(BindingInfoRegistry bindingInfoRegistry) {
-        this.bindingInfoRegistry = null;
-    }
-
-    /**
-     * Sets the thing registry.
-     *
-     * @param thingRegistry the non-null thing registry
-     */
-    @Reference
-    public void setThingRegistry(ThingRegistry thingRegistry) {
-        Objects.requireNonNull(thingRegistry, "thingRegistry cannot be null");
-        this.thingRegistry = thingRegistry;
-    }
-
-    /**
-     * Unset thing registry.
-     *
-     * @param thingRegistry the thing registry (ignored)
-     */
-    public void unsetThingRegistry(ThingRegistry thingRegistry) {
-        this.thingRegistry = null;
-    }
-
-    /**
-     * Sets the thing type registry.
-     *
-     * @param thingTypeRegistry the non-null thing type registry
-     */
-    @Reference
-    public void setThingTypeRegistry(ThingTypeRegistry thingTypeRegistry) {
-        Objects.requireNonNull(thingTypeRegistry, "thingTypeRegistry cannot be null");
-        this.thingTypeRegistry = thingTypeRegistry;
-    }
-
-    /**
-     * Unset thing type registry.
-     *
-     * @param thingTypeRegistry the thing type registry (ignored)
-     */
-    public void unsetThingTypeRegistry(ThingTypeRegistry thingTypeRegistry) {
-        this.thingTypeRegistry = null;
-    }
-
-    /**
-     * Sets the item channel link registry.
-     *
-     * @param itemChannelLinkRegistry the non-null item channel link registry
-     */
-    @Reference
-    public void setItemChannelLinkRegistry(ItemChannelLinkRegistry itemChannelLinkRegistry) {
-        Objects.requireNonNull(itemChannelLinkRegistry, "itemChannelLinkRegistry cannot be null");
-        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
-    }
-
-    /**
-     * Unset item channel link registry.
-     *
-     * @param itemChannelLinkRegistry the item channel link registry (ignored)
-     */
-    public void unsetItemChannelLinkRegistry(ItemChannelLinkRegistry itemChannelLinkRegistry) {
-        this.itemChannelLinkRegistry = null;
-    }
-
-    /**
-     * Sets the channel type registry.
-     *
-     * @param channelTypeRegistry the non-null channel type registry
-     */
-    @Reference
-    public void setChannelTypeRegistry(ChannelTypeRegistry channelTypeRegistry) {
-        Objects.requireNonNull(channelTypeRegistry, "channelTypeRegistry cannot be null");
-        this.channelTypeRegistry = channelTypeRegistry;
-    }
-
-    /**
-     * Unset channel type registry.
-     *
-     * @param channelTypeRegistry the channel type registry (ignored)
-     */
-    public void unsetChannelTypeRegistry(ChannelTypeRegistry channelTypeRegistry) {
-        this.channelTypeRegistry = null;
-    }
-
-    /**
-     * Sets the MDNS client.
-     *
-     * @param mdnsClient the non-null MDNS client
-     */
-    @Reference
-    public void setMDNSClient(MDNSClient mdnsClient) {
-        Objects.requireNonNull(mdnsClient, "mdnsClient cannot be null");
-        this.mdnsClient = mdnsClient;
-    }
-
-    /**
-     * Unset MDNS client.
-     *
-     * @param mdnsClient the mdns client (ignored)
-     */
-    public void unsetMDNSClient(MDNSClient mdnsClient) {
-        this.mdnsClient = null;
-    }
-
-    /**
-     * Sets the event publisher.
-     *
-     * @param eventPublisher the new event publisher
-     */
-    @Reference
-    public void setEventPublisher(EventPublisher eventPublisher) {
-        Objects.requireNonNull(eventPublisher, "eventPublisher cannot be null");
-        this.eventPublisher = eventPublisher;
-    }
-
-    /**
-     * Unset event publisher.
-     *
-     * @param eventPublisher the event publisher (ignored)
-     */
-    public void unsetEventPublisher(EventPublisher eventPublisher) {
-        this.eventPublisher = null;
-    }
-
-    /**
-     * Sets the network address service
-     *
-     * @param networkAddressService the network address service
-     */
-    @Reference
-    public void setNetworkAddressService(NetworkAddressService networkAddressService) {
-        Objects.requireNonNull(networkAddressService, "networkAddressService cannot be null");
-        this.networkAddressService = networkAddressService;
-        networkAddressService.addNetworkAddressChangeListener(this);
-    }
-
-    /**
-     * Unsets network address service
-     *
-     * @param networkAddressService address service
-     */
-    public void unsetNetworkAddressService(NetworkAddressService networkAddressService) {
-        networkAddressService.removeNetworkAddressChangeListener(this);
-        this.networkAddressService = null;
-    }
-
-    /**
-     * Activates this service. The activation will start up the brain discovery service and register the dashboard tile
-     *
-     * @param componentContext the non-null component context
-     */
     @Activate
-    public void activate(final ComponentContext componentContext) {
-        Objects.requireNonNull(componentContext, "componentContext cannot be null");
+    public NeeoService(ComponentContext componentContext, @Reference HttpService httpService,
+            @Reference ItemRegistry itemRegistry, @Reference ThingRegistry thingRegistry,
+            @Reference BindingInfoRegistry bindingInfoRegistry, @Reference ChannelTypeRegistry channelTypeRegistry,
+            @Reference ThingTypeRegistry thingTypeRegistry, @Reference ItemChannelLinkRegistry itemChannelLinkRegistry,
+            @Reference MDNSClient mdnsClient, @Reference EventPublisher eventPublisher,
+            @Reference NetworkAddressService networkAddressService, @Reference ClientBuilder clientBuilder) {
+        this.httpService = httpService;
+        this.itemRegistry = itemRegistry;
+        this.bindingInfoRegistry = bindingInfoRegistry;
+        this.channelTypeRegistry = channelTypeRegistry;
+        this.thingRegistry = thingRegistry;
+        this.thingTypeRegistry = thingTypeRegistry;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+        this.mdnsClient = mdnsClient;
+        this.eventPublisher = eventPublisher;
+        this.networkAddressService = networkAddressService;
+        this.clientBuilder = clientBuilder;
 
         logger.debug("Neeo Service activated");
         final ServiceContext localContext = new ServiceContext(componentContext, validate(httpService, "httpService"),
@@ -378,7 +166,7 @@ public class NeeoService implements EventSubscriber, NetworkAddressChangeListene
                 validate(eventPublisher, "eventPublisher"), validate(networkAddressService, "networkAddressService"));
 
         context = localContext;
-        discovery = new MdnsBrainDiscovery(localContext);
+        discovery = new MdnsBrainDiscovery(localContext, clientBuilder);
         discovery.addListener(discoveryListener);
 
         try {
@@ -401,6 +189,28 @@ public class NeeoService implements EventSubscriber, NetworkAddressChangeListene
             localDiscovery.startDiscovery();
         }
     }
+
+    /**
+     * The event filter to apply to this service
+     */
+    private final EventFilter eventFilter = event -> {
+        logger.trace("apply: {}", event);
+
+        for (NeeoBrainServlet ns : servlets) {
+            final List<EventFilter> efs = ns.getEventFilters();
+            if (efs != null) {
+                for (EventFilter ef : efs) {
+                    if (ef.apply(event)) {
+                        logger.trace("apply (true): {}", event);
+                        return true;
+                    }
+                }
+            }
+        }
+
+        logger.trace("apply (false): {}", event);
+        return false;
+    };
 
     /**
      * Helper method to validate that the specific item wasn't null and convert it's type to a non-nullable type
@@ -441,8 +251,6 @@ public class NeeoService implements EventSubscriber, NetworkAddressChangeListene
                 NeeoUtil.close(servlet);
             }
             servlets.clear();
-
-            context = null;
         }
 
         if (dashboardServlet != null) {
@@ -472,7 +280,7 @@ public class NeeoService implements EventSubscriber, NetworkAddressChangeListene
                         servletUrl);
                 try {
                     final NeeoBrainServlet newServlet = NeeoBrainServlet.create(localContext, servletUrl,
-                            sysInfo.getHostname(), ipAddress);
+                            sysInfo.getHostname(), ipAddress, clientBuilder);
                     servlets.add(newServlet);
 
                     localContext.getHttpService().registerServlet(servletUrl, newServlet, new Hashtable<>(),

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoApi.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoApi.java
@@ -27,6 +27,8 @@ import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 
+import javax.ws.rs.client.ClientBuilder;
+
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
@@ -68,6 +70,8 @@ public class NeeoApi implements AutoCloseable {
     /** The brain's IP address */
     private final String brainIpAddress;
 
+    private final ClientBuilder clientBuilder;
+
     /** The URL of the brain */
     private final String brainUrl;
 
@@ -107,8 +111,7 @@ public class NeeoApi implements AutoCloseable {
     private final AtomicReference<@Nullable ScheduledFuture<?>> checkStatus = new AtomicReference<>(null);
 
     /** The {@link HttpRequest} used for making requests */
-    private final AtomicReference<HttpRequest> request = new AtomicReference<>(new HttpRequest());
-
+    private final AtomicReference<HttpRequest> request;
     /** Whether the brain is currently connected */
     private final AtomicBoolean connected = new AtomicBoolean(false);
 
@@ -120,21 +123,25 @@ public class NeeoApi implements AutoCloseable {
      * @param context the non-null {@link ServiceContext}
      * @throws IOException if an exception occurs connecting to the brain
      */
-    public NeeoApi(String ipAddress, String brainId, ServiceContext context) throws IOException {
+    public NeeoApi(String ipAddress, String brainId, ServiceContext context, ClientBuilder clientBuilder)
+            throws IOException {
         NeeoUtil.requireNotEmpty(ipAddress, "ipAddress cannot be empty");
         NeeoUtil.requireNotEmpty(brainId, "brainId cannot be empty");
         Objects.requireNonNull(context, "context cannot be null");
 
         this.brainIpAddress = ipAddress;
         this.brainId = brainId;
+        this.clientBuilder = clientBuilder;
         this.brainUrl = NeeoConstants.PROTOCOL + (ipAddress.startsWith("/") ? ipAddress.substring(1) : ipAddress) + ":"
                 + NeeoConstants.DEFAULT_BRAIN_PORT;
-        deviceKeys = new NeeoDeviceKeys(brainUrl);
+        deviceKeys = new NeeoDeviceKeys(brainUrl, clientBuilder);
 
-        this.systemInfo = getSystemInfo(ipAddress);
+        request = new AtomicReference<>(new HttpRequest(clientBuilder));
+
+        this.systemInfo = getSystemInfo(ipAddress, clientBuilder);
 
         String name = brainId;
-        try (HttpRequest request = new HttpRequest()) {
+        try (HttpRequest request = new HttpRequest(clientBuilder)) {
             logger.debug("Getting existing device mappings from {}{}", brainUrl, NeeoConstants.PROJECTS_HOME);
             final HttpResponse resp = request.sendGetCommand(brainUrl + NeeoConstants.PROJECTS_HOME);
             if (resp.getHttpCode() != HttpStatus.OK_200) {
@@ -196,12 +203,12 @@ public class NeeoApi implements AutoCloseable {
      * @return the non-null {@link NeeoSystemInfo} for the address
      * @throws IOException Signals that an I/O exception has occurred or the URL is not a brain
      */
-    public static NeeoSystemInfo getSystemInfo(String ipAddress) throws IOException {
+    public static NeeoSystemInfo getSystemInfo(String ipAddress, ClientBuilder clientBuilder) throws IOException {
         NeeoUtil.requireNotEmpty(ipAddress, "ipAddress cannot be empty");
         final String sysInfo = NeeoConstants.PROTOCOL + (ipAddress.startsWith("/") ? ipAddress.substring(1) : ipAddress)
                 + ":" + NeeoConstants.DEFAULT_BRAIN_PORT + NeeoConstants.SYSTEMINFO;
 
-        try (HttpRequest req = new HttpRequest()) {
+        try (HttpRequest req = new HttpRequest(clientBuilder)) {
             final HttpResponse res = req.sendGetCommand(sysInfo);
             if (res.getHttpCode() == HttpStatus.OK_200) {
                 return Objects.requireNonNull(GSON.fromJson(res.getContent(), NeeoSystemInfo.class));
@@ -387,7 +394,7 @@ public class NeeoApi implements AutoCloseable {
         try {
             setConnected(false);
 
-            NeeoUtil.close(request.getAndSet(new HttpRequest()));
+            NeeoUtil.close(request.getAndSet(new HttpRequest(clientBuilder)));
 
             NeeoUtil.checkInterrupt();
             registerApi();

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoBrainServlet.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoBrainServlet.java
@@ -16,6 +16,8 @@ import java.io.IOException;
 import java.net.InetAddress;
 import java.util.Objects;
 
+import javax.ws.rs.client.ClientBuilder;
+
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.openhab.io.neeo.internal.models.BrainStatus;
 import org.openhab.io.neeo.internal.servletservices.NeeoBrainSearchService;
@@ -44,8 +46,9 @@ public class NeeoBrainServlet extends AbstractServlet {
      * @param servletUrl the non-null, non-empty servlet URL
      * @param api the non-null API
      */
-    private NeeoBrainServlet(ServiceContext context, String servletUrl, NeeoApi api) {
-        super(context, servletUrl, new NeeoBrainSearchService(context), new NeeoBrainService(api, context));
+    private NeeoBrainServlet(ServiceContext context, String servletUrl, NeeoApi api, ClientBuilder clientBuilder) {
+        super(context, servletUrl, new NeeoBrainSearchService(context),
+                new NeeoBrainService(api, context, clientBuilder));
 
         Objects.requireNonNull(context, "context cannot be null");
         NeeoUtil.requireNotEmpty(servletUrl, "servletUrl cannot be empty");
@@ -65,16 +68,16 @@ public class NeeoBrainServlet extends AbstractServlet {
      * @throws IOException when an exception occurs contacting the brain
      */
     public static NeeoBrainServlet create(ServiceContext context, String servletUrl, String brainId,
-            InetAddress ipAddress) throws IOException {
+            InetAddress ipAddress, ClientBuilder clientBuilder) throws IOException {
         Objects.requireNonNull(context, "context cannot be null");
         NeeoUtil.requireNotEmpty(servletUrl, "servletUrl cannot be empty");
         NeeoUtil.requireNotEmpty(brainId, "brainId cannot be empty");
         Objects.requireNonNull(ipAddress, "ipAddress cannot be null");
 
-        final NeeoApi api = new NeeoApi(ipAddress.getHostAddress(), brainId, context);
+        final NeeoApi api = new NeeoApi(ipAddress.getHostAddress(), brainId, context, clientBuilder);
         api.start();
 
-        return new NeeoBrainServlet(context, servletUrl, api);
+        return new NeeoBrainServlet(context, servletUrl, api, clientBuilder);
     }
 
     /**

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoDeviceKeys.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/NeeoDeviceKeys.java
@@ -21,6 +21,8 @@ import java.util.Objects;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 
+import javax.ws.rs.client.ClientBuilder;
+
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jetty.http.HttpStatus;
@@ -52,15 +54,18 @@ public class NeeoDeviceKeys {
     /** The brain's url */
     private final String brainUrl;
 
+    private final ClientBuilder clientBuilder;
+
     /**
      * Creates the object from the context and brainUrl
      *
      * @param brainUrl the non-empty brain url
      */
-    NeeoDeviceKeys(String brainUrl) {
+    NeeoDeviceKeys(String brainUrl, ClientBuilder clientBuilder) {
         NeeoUtil.requireNotEmpty(brainUrl, "brainUrl cannot be empty");
 
         this.brainUrl = brainUrl;
+        this.clientBuilder = clientBuilder;
     }
 
     /**
@@ -69,7 +74,7 @@ public class NeeoDeviceKeys {
      * @throws IOException Signals that an I/O exception has occurred.
      */
     void refresh() throws IOException {
-        try (HttpRequest request = new HttpRequest()) {
+        try (HttpRequest request = new HttpRequest(clientBuilder)) {
             logger.debug("Getting existing device mappings from {}{}", brainUrl, NeeoConstants.PROJECTS_HOME);
             final HttpResponse resp = request.sendGetCommand(brainUrl + NeeoConstants.PROJECTS_HOME);
             if (resp.getHttpCode() != HttpStatus.OK_200) {

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/discovery/MdnsBrainDiscovery.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/discovery/MdnsBrainDiscovery.java
@@ -34,6 +34,7 @@ import java.util.stream.Collectors;
 import javax.jmdns.ServiceEvent;
 import javax.jmdns.ServiceInfo;
 import javax.jmdns.ServiceListener;
+import javax.ws.rs.client.ClientBuilder;
 
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNullByDefault;
@@ -106,14 +107,17 @@ public class MdnsBrainDiscovery extends AbstractBrainDiscovery {
     /** The file we store definitions in */
     private final File file = new File(NeeoConstants.FILENAME_DISCOVEREDBRAINS);
 
+    private final ClientBuilder clientBuilder;
+
     /**
      * Creates the MDNS brain discovery from the given {@link ServiceContext}
      *
      * @param context the non-null service context
      */
-    public MdnsBrainDiscovery(ServiceContext context) {
+    public MdnsBrainDiscovery(ServiceContext context, ClientBuilder clientBuilder) {
         Objects.requireNonNull(context, "context cannot be null");
         this.context = context;
+        this.clientBuilder = clientBuilder;
     }
 
     /**
@@ -250,7 +254,7 @@ public class MdnsBrainDiscovery extends AbstractBrainDiscovery {
 
         NeeoSystemInfo sysInfo;
         try {
-            sysInfo = NeeoApi.getSystemInfo(brainInfo.getValue().toString());
+            sysInfo = NeeoApi.getSystemInfo(brainInfo.getValue().toString(), clientBuilder);
         } catch (IOException e) {
             // We can get an MDNS notification BEFORE the brain is ready to process.
             // if that happens, we'll get an IOException (usually bad gateway message), schedule another attempt to get
@@ -299,7 +303,7 @@ public class MdnsBrainDiscovery extends AbstractBrainDiscovery {
 
         try {
             final InetAddress addr = InetAddress.getByName(ipAddress);
-            final NeeoSystemInfo sysInfo = NeeoApi.getSystemInfo(ipAddress);
+            final NeeoSystemInfo sysInfo = NeeoApi.getSystemInfo(ipAddress, clientBuilder);
             logger.debug("Manually adding brain ({}) with system information: {}", ipAddress, sysInfo);
 
             systemsLock.lock();

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/net/HttpRequest.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/net/HttpRequest.java
@@ -47,8 +47,8 @@ public class HttpRequest implements AutoCloseable {
     /**
      * Instantiates a new request
      */
-    public HttpRequest() {
-        client = ClientBuilder.newClient();
+    public HttpRequest(ClientBuilder clientBuilder) {
+        client = clientBuilder.build();
 
         if (logger.isDebugEnabled()) {
             client.register(new LoggingFilter(new Slf4LoggingAdapter(logger), true));

--- a/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/NeeoBrainService.java
+++ b/bundles/org.openhab.io.neeo/src/main/java/org/openhab/io/neeo/internal/servletservices/NeeoBrainService.java
@@ -21,6 +21,7 @@ import java.util.concurrent.ScheduledExecutorService;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.client.ClientBuilder;
 
 import org.apache.commons.lang.StringUtils;
 import org.eclipse.jdt.annotation.NonNull;
@@ -90,7 +91,7 @@ public class NeeoBrainService extends DefaultServletService {
     private final ServiceContext context;
 
     /** The HTTP request */
-    private final HttpRequest request = new HttpRequest();
+    private final HttpRequest request;
 
     /** The scheduler to use to schedule recipe execution */
     private final ScheduledExecutorService scheduler = ThreadPoolManager
@@ -114,7 +115,7 @@ public class NeeoBrainService extends DefaultServletService {
      * @param api the non-null api
      * @param context the non-null context
      */
-    public NeeoBrainService(NeeoApi api, ServiceContext context) {
+    public NeeoBrainService(NeeoApi api, ServiceContext context, ClientBuilder clientBuilder) {
         Objects.requireNonNull(api, "api cannot be null");
         Objects.requireNonNull(context, "context cannot be null");
 
@@ -125,6 +126,7 @@ public class NeeoBrainService extends DefaultServletService {
         scheduler.execute(() -> {
             resendState();
         });
+        request = new HttpRequest(clientBuilder);
     }
 
     /**

